web3.js/src/transaction.js

```diff
@@ -339,9 +355,24 @@ export class Transaction {
   }
 
   /**
-   * Sign the Transaction with the specified accounts.  Multiple signatures may
+   * Specify the public keys which will be used to sign the Transaction.
+   * The first signer will be used as the transaction fee payer account.
+   *
+   * Signatures can be added with either `partialSign` or `addSignature`
+   */
+  setSigners(...signers: Array<PublicKey>) {
```

Comment:
I like this interface better!  If the whole "Make sure you put the fee payer first" thing doesn't work out, we can just make it explicit here.

How do we ensure that `signers` is well-formed?  `this.signatures` needs to be ordered the same as the `Message.accountKeys`.  I'm not sure we can know this without having called `compileMessage()` 🤔

Reply:
This has the same problem as the original `sign` method right? I don't think it's a new issue. Can we punt on that?

Reply:
Created an issue here: https://github.com/solana-labs/solana/issues/12155

Reply:
I'm pretty sure signatures order was correct before, https://github.com/solana-labs/solana/pull/12109/files#diff-86eea70a26ecf01c3229f804beb21ac4L297-L299